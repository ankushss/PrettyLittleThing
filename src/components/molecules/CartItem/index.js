import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {useDispatch} from 'react-redux';
import {addToCart, removeFromCart} from '../../../store/cart';
import {Counter} from '../../atoms';

export const CartItem = ({item}) => {
  const dispatch = useDispatch();

  const onPressAdd = () => {
    dispatch(addToCart(item));
  };

  const onPressRemove = () => {
    dispatch(removeFromCart(item));
  };

  return (
    <View style={styles.container}>
      <Image source={{uri: item.img}} style={styles.image} />
      <View style={styles.textContainer}>
        <Text numberOfLines={2}>{item.name}</Text>
        <Text style={styles.price}>${item.price}</Text>
        <Text>
          Color: <Text style={styles.colorText}>{item.colour}</Text>
        </Text>
        <View style={styles.counterContainer}>
          <Counter
            value={item.quantity}
            onPressAdd={onPressAdd}
            onPressRemove={onPressRemove}
          />
          <Text>
            Total:{' '}
            <Text style={styles.totalText}>${item.quantity * item.price}</Text>
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  image: {
    width: 80,
    height: (1180 / 740) * 80,
    resizeMode: 'contain',
    marginRight: 10,
  },
  textContainer: {
    flex: 1,
  },
  price: {
    fontWeight: 'bold',
    fontSize: 20,
    marginTop: 4,
  },
  colorText: {
    fontWeight: 'bold',
  },
  button: {
    marginTop: 10,
  },
  counterContainer: {
    marginTop: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  totalText: {
    fontWeight: 'bold',
  },
});
