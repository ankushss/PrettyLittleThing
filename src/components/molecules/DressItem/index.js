import React from 'react';
import {Image, StyleSheet, Text, useWindowDimensions, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {addToCart, getItemQuantityInCart} from '../../../store/cart';

import {Button} from '../../atoms';

export const DressItem = ({item, index}) => {
  const dimensions = useWindowDimensions();
  const dispatch = useDispatch();
  const itemQuantityInCart = useSelector(getItemQuantityInCart(item.id));
  const isItemAddedToCard = itemQuantityInCart > 0;

  const onPressAddToCard = () => {
    dispatch(addToCart(item));
  };

  return (
    <View style={styles.container(dimensions, index)}>
      <Image source={{uri: item.img}} style={styles.image(dimensions)} />
      <Text numberOfLines={2}>{item.name}</Text>
      <Text style={styles.price}>${item.price}</Text>
      <Text>
        Color: <Text style={styles.colorText}>{item.colour}</Text>
      </Text>
      <Button
        testID="addToCartButton"
        variant={isItemAddedToCard ? 'outlined' : 'contained'}
        style={styles.button}
        onPress={onPressAddToCard}
        disabled={isItemAddedToCard}>
        {isItemAddedToCard ? 'ADDED TO CART' : 'ADD TO CART'}
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: (dimensions, index) => ({
    width: (dimensions.width - 60) / 2,
    marginRight: index % 2 === 0 ? 20 : 0,
    marginBottom: 20,
  }),
  image: dimensions => ({
    width: (dimensions.width - 60) / 2,
    height: ((1180 / 740) * (dimensions.width - 60)) / 2,
    resizeMode: 'contain',
    marginBottom: 10,
  }),
  price: {
    fontWeight: 'bold',
    fontSize: 20,
    marginTop: 4,
  },
  colorText: {
    fontWeight: 'bold',
  },
  button: {
    marginTop: 10,
  },
});
