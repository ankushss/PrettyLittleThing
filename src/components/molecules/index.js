export * from './DressItem';
export * from './CartItem';
