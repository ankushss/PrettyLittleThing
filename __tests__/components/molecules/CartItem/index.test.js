import React from 'react';
import {create} from 'react-test-renderer';
import {CartItem} from '../../../../src/components/molecules';

const mockItem = {
  id: 1,
  colour: 'mockColor',
  name: 'mockName',
  price: 1,
  img: 'mockImgUrl',
  quantity: 1,
};

describe('CartItem', () => {
  it('matches snapshot', () => {
    const component = create(<CartItem item={mockItem} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('onPress Add', () => {
    const component = create(<CartItem item={mockItem} />);
    const addButton = component.root.findByProps({testID: 'addButton'});
    addButton.props.onPress();
  });

  it('onPress Remove', () => {
    const component = create(<CartItem item={mockItem} />);
    expect(component.toJSON()).toMatchSnapshot();
    const removeButton = component.root.findByProps({testID: 'removeButton'});
    removeButton.props.onPress();
  });
});
