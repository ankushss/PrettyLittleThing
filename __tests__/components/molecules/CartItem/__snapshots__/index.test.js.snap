// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "mockImgUrl",
      }
    }
    style={
      Object {
        "height": 127.56756756756756,
        "marginRight": 10,
        "resizeMode": "contain",
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={2}
    >
      mockName
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 4,
        }
      }
    >
      $
      1
    </Text>
    <Text>
      Color: 
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        mockColor
      </Text>
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="addButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "borderRadius": 100,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <Text>
              +
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 10,
            }
          }
        >
          1
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="removeButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "borderRadius": 100,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <Text>
              -
            </Text>
          </View>
        </View>
      </View>
      <Text>
        Total:
         
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          $
          1
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`CartItem onPress Remove 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "mockImgUrl",
      }
    }
    style={
      Object {
        "height": 127.56756756756756,
        "marginRight": 10,
        "resizeMode": "contain",
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      numberOfLines={2}
    >
      mockName
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 4,
        }
      }
    >
      $
      1
    </Text>
    <Text>
      Color: 
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        mockColor
      </Text>
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="addButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "borderRadius": 100,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <Text>
              +
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 10,
            }
          }
        >
          1
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="removeButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "borderRadius": 100,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <Text>
              -
            </Text>
          </View>
        </View>
      </View>
      <Text>
        Total:
         
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          $
          1
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
